src/qt/nametablemodel.cpp

```diff
@@ -0,0 +1,319 @@
+    return columns.length();
+}
+
+QVariant NameTableModel::data(const QModelIndex &index, int role) const
+{
+    if(!index.isValid())
+        return QVariant();
+
+    NameTableEntry *rec = static_cast<NameTableEntry*>(index.internalPointer());
+
+    // TODO: implement Qt::ForegroudRole for font color styling for states?
+    // TODO: implement Qt::ToolTipRole show name status on tooltip
+    if(role == Qt::DisplayRole || role == Qt::EditRole)
+    {
+        switch(index.column())
+        {
+            case Name:
+                return rec->name;
+            case Value:
+                return rec->value;
+            case ExpiresIn:
+                return rec->expiresIn;
+            case NameStatus:
+                return rec->nameStatus;
+        }
+    }
+
+    if(role == Qt::TextAlignmentRole)
+        return column_alignments[index.column()];
+
+    return QVariant();
+}
+
+QVariant NameTableModel::headerData(int section, Qt::Orientation orientation, int role) const
+{
+    if(orientation != Qt::Horizontal)
+        return QVariant();
+
+    if(role == Qt::DisplayRole)
+        return columns[section];
+
+    if(role == Qt::TextAlignmentRole)
+        return column_alignments[section];
+
+    if(role == Qt::ToolTipRole)
+    {
+        switch(section)
+        {
+            case Name:
+                return tr("Name registered using Namecoin.");
+
+            case Value:
+                return tr("Data associated with the name.");
+
+            case ExpiresIn:
+                return tr("Number of blocks, after which the name will expire. Update name to renew it.\nEmpty cell means pending(awaiting automatic name_firstupdate or awaiting network confirmation).");
+        }
+    }
+    return QVariant();
+}
+
+Qt::ItemFlags NameTableModel::flags(const QModelIndex &index) const
+{
+    if(!index.isValid())
+        return 0;
+
+    return Qt::ItemIsSelectable | Qt::ItemIsEnabled;
+}
+
+QModelIndex NameTableModel::index(int row, int column, const QModelIndex &parent /* = QModelIndex()*/) const
+{
+    Q_UNUSED(parent);
+    NameTableEntry *data = priv->index(row);
+    if(data)
+    {
+        return createIndex(row, column, priv->index(row));
+    }
+    return QModelIndex();
+}
+
+void
+NameTableModel::emitDataChanged(int idx)
+{
+    //emit
+    dataChanged(index(idx, 0), index(idx, columns.length()-1));
+}
+
+void
+NameTableModel::subscribeToCoreSignals()
+{
+    // Connect signals to wallet
+    //m_handler_transaction_changed = walletModel->wallet().handleTransactionChanged(std::bind(NotifyTransactionChanged, this, std::placeholders::_1, std::placeholders::_2));
+    //m_handler_show_progress = walletModel->wallet().handleShowProgress(std::bind(ShowProgress, this, std::placeholders::_1, std::placeholders::_2));
+}
+
+void
+NameTableModel::unsubscribeFromCoreSignals()
+{
+    // Disconnect signals from wallet
+    //m_handler_transaction_changed->disconnect();
+    //m_handler_show_progress->disconnect();
+}
```

Comment:
Why not just wipe these out if we're not using this anymore? I think we should be good to go with the refreshing of the whole name table being triggered from `updateExpiration` and `processNewTransaction`.

Reply:
Thanks, I've removed this (along with a bit of other dead code).